vaadin-checkbox-flow/src/main/java/com/vaadin/flow/component/checkbox/GeneratedVaadinCheckbox.java

```diff
@@ -347,32 +347,63 @@ protected Registration addIndeterminateChangeListener(
             SerializableFunction<T, P> modelToPresentation) {
         super("checked", defaultValue, elementPropertyType, presentationToModel,
                 modelToPresentation);
-        if (initialValue != null) {
-            setModelValue(initialValue, false);
+        if (getElement().getProperty("checked") == null
```

Comment:
I suppose we don't need `getElement().getProperty("checked") == null` here, because init value is in arguments list and no optional value flag provided

Reply:
Leftover, thank you.